prog/vnet/update_load_balancer.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+class Prog::Vnet::UpdateLoadBalancer < Prog::Base
+  subject_is :vm
+
+  def load_balancer
+    @load_balancer ||= LoadBalancer[frame.fetch("load_balancer_id")]
+  end
+
+  label def update_load_balancer
+    vm.vm_host.sshable.cmd("sudo ip netns exec #{vm.inhost_name} nft --file -", stdin: generate_lb_based_nat_rules)
+
+    pop "load balancer is updated"
+  end
+
+  label def remove_load_balancer
+    vm.vm_host.sshable.cmd("sudo ip netns exec #{vm.inhost_name} nft --file -", stdin: generate_nat_rules(vm.ephemeral_net4.to_s, vm.nics.first.private_ipv4.network.to_s))
+    pop "load balancer is updated"
+  end
+
+  def generate_lb_based_nat_rules
+    public_ipv4 = vm.ephemeral_net4.to_s
+    public_ipv6 = vm.ephemeral_net6.to_s
+    private_ipv4 = vm.nics.first.private_ipv4.network
+    private_ipv6 = vm.nics.first.private_ipv6.nth(2)
```

Comment:
This can't be sound, can it? since `nics` is unordered and you take first. I would check it for length one (i.e. `first` is unambiguous) and crash, because whenever someone (you?) comes along to add multiple-nic support to a VM, it's going to be hard to find/remember this code.

Reply:
I'm not sure what you mean by "for length one". You're right, using an unordered "nics" call is an invitation to failure the moment we add multiple nics support. I'll fix that in the next iteration in the whole code utilising the nic.created_at column. Since we have these "vm.nics.first" call everywhere in the code, I'll skip here, too.